### PR TITLE
[bug 1144159] Remove usage of django-celery.

### PR DESCRIPTION
**Pull request by rlr**
* Use celery the way suggested in the docs.
* Create a celeryd command so our infra keeps working without changes.
* Move some monkeypatching stuff around.

There are some things that need to be cleaned up. Wanted to posted it like this so we can discuss. I'll add some comments inline.

f?

### CODE REVIEW

File: fjord/__init__.py [hunk 1]
```diff
@@ -1,7 +1,16 @@
+# First we import absolute imports from the future, so that our celery.py
+# module will not clash with the library
+from __future__ import absolute_import
+
 import os
 import site
 import sys
 
+from fjord.manage_utils import monkeypatch
```
Comment (rlr) [inline fjord/__init__.py]: ugh. so celery needs the monkeypatching to be in place too. Should we just move all the manage_utils stuff here?

File: fjord/__init__.py [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'fjord.settings')
```
Comment (rlr) [inline fjord/__init__.py]: this is probably in like 3 places now.

File: fjord/base/management/commands/celeryd.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+from fjord.celery import app
+
+
+worker = worker.worker(app=app)
+
+
+# This code is based on django-celery code. It's a merge of
+# djcelery.management.base.CeleryCommand and
+# djcelery.management.commands.celeryd.Command.
```
Comment (rlr) [inline fjord/base/management/commands/celeryd.py]: It's possible some of the code in here below isn't needed.

File: fjord/celery.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+# Using a string here means the worker will not have to
+# pickle the object when using Windows.
+app.config_from_object('django.conf:settings')
+app.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
+
+
+@app.task(bind=True)
+def debug_task(self):
+    print('Request: {0!r}'.format(self.request))
```
Comment (rlr) [inline fjord/celery.py]: this was cool to have for quick tests. Not sure if it's useful going forward.

File: fjord/__init__.py (hunk 1 above)
Comment (rlr) [inline fjord/__init__.py]: To be clear, this is probably not needed if I do `./manage.py celeryd`. But if I do `celery -A fjord worker -l info` it is. And that's the future!

File: fjord/__init__.py (hunk 2 above)
Comment (willkg) [inline fjord/__init__.py]: Only three? We should do better!

File: fjord/celery.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from __future__ import absolute_import
+
+import os
+
+from celery import Celery
+
+# set the default Django settings module for the 'celery' program.
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'fjord.settings')
```
Comment (willkg) [inline fjord/celery.py]: You probably don't need this one given that in order to import this module, you first have to import `fjord/__init__.py` which already has one of these.

File: fjord/celery.py (hunk 4 above)
Comment (willkg) [inline fjord/celery.py]: What's this?

File: fjord/manage_utils.py [hunk 6]
```diff
@@ -63,6 +61,9 @@ def check_dependencies():
     from pip.req import parse_requirements
     from pip.download import PipSession
 
+    # Avoid circular import
+    from fjord import path
```
Comment (willkg) [inline fjord/manage_utils.py]: `path` is such a goofy thing. It pains me to have circular imports for something that's such a trivial and non-interesting bit of code. We should figure out what to do with `ROOT` and `path`.

File: fjord/base/management/commands/celeryd.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+from django.core.management.base import BaseCommand
+
+import celery
+from celery.bin import worker
+
+from fjord.celery import app
+
+
+worker = worker.worker(app=app)
```
Comment (willkg) [inline fjord/base/management/commands/celeryd.py]: `worker` is a module, but then this line changes it into something else. Is there a better way to do this that doesn't involve overloading "worker"?

File: fjord/base/management/commands/celeryd.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    help = 'Old alias to the "celery worker" command.'
+    options = (BaseCommand.option_list
+               + worker.get_options()
+               + worker.preload_options)
+    skip_opts = ['--app', '--loader', '--config', '--no-color']
+    requires_model_validation = False
+    keep_base_opts = False
+    stdout, stderr = sys.stdout, sys.stderr
+
+
```
Comment (willkg) [inline fjord/base/management/commands/celeryd.py]: Nit: there should be only one blank line here.

File: fjord/__init__.py [hunk 9]
```diff
@@ -49,3 +58,11 @@ def setup_vendor_path():
 
 
 setup_vendor_path()
+
+monkeypatch()
```
Comment (willkg) [inline fjord/__init__.py]: Why do we have to do the `monkeypatch()` call here? Which code path doesn't have monkeypatching?

File: fjord/celery.py (hunk 4 above)
Comment (rlr) [inline fjord/celery.py]: Oh, it's copy paste from the first step docs. I was using it to test celery from the shell.

```
In [1]: from fjord.celery import debug_task

In [2]: debug_task.delay()
Out[2]: <AsyncResult: 1ec26e79-ca12-49c4-b30c-e6bde4c49a5d>

In [3]: exit()
```

File: fjord/celery.py (hunk 5 above)
Comment (rlr) [inline fjord/celery.py]: ha! forgot about his one!

File: fjord/__init__.py (hunk 2 above)
Comment (rlr) [inline fjord/__init__.py]: The code path of using the celery command as `$ celery -A fjord worker -l info` which is the suggested way now.

Comment (rlr) [inline fjord/__init__.py]: Note, that doesn't work in the vendor setup. I had to create a new virtualenv with requirements installed in it to test this path. I guess it should work in vagrant too.

File: fjord/__init__.py (hunk 1 above)
Comment (willkg) [inline fjord/__init__.py]: I agree we want to switch to this, but not until after we ditch `vendor/`. So I vote we keep things simple for now and not do the monkeypatch thing here.

File: fjord/__init__.py [hunk 10]
```diff
@@ -49,3 +58,11 @@ def setup_vendor_path():
...
+
+
+# Setup celery
+# This will make sure the app is always imported when
+# Django starts so that shared_task will use this app.
+from .celery import app as celery_app
```
Comment (willkg) [inline fjord/__init__.py]: I'm pretty sure we only need this for `celery -A ...` launching. Let's nix this for now.

**Comment by willkg** [comment]
I created bug #1161236 to cover switching to "celery -A ...".

https://bugzilla.mozilla.org/show_bug.cgi?id=1161236

In the meantime, let's simplify this PR to just the bits we need to ditch djcelery and have our own celeryd Django command. That makes it easier to maintain for the next x months until we're in AWS land.

**Comment by rlr** [comment]
I think that covers all the feedback so far :point_up:

**Comment by willkg** [comment]
This looks good to me. Let's test it on -stage! r+

**Pull request merged**